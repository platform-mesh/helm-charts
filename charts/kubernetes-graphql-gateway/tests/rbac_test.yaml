suite: Test the Template files
values:
- ../values.yaml
- ../test_values.yaml
release:
  name: platform-mesh
  namespace: platform-mesh-system
chart:
  appVersion: 1.0.0
templates:
  - clusterrole.yaml
  - clusterrolebinding.yaml
tests:
  - it: matches the Snapshot
    asserts:
    - matchSnapshot: {}
  - it: matches the Snapshot with internal rbac creation enabled
    set:
      kubeConfig.enabled: false
      kcp.enabled: false
    asserts:
      - matchSnapshot: {}