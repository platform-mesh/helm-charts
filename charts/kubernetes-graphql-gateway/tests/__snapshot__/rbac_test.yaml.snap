matches the Snapshot with internal rbac creation enabled:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: platform-mesh-kubernetes-graphql-gateway
      namespace: platform-mesh-system
    rules:
      - apiGroups:
          - ""
        resources:
          - users
          - serviceaccounts
        verbs:
          - impersonate
      - apiGroups:
          - '*'
        resources:
          - '*'
        verbs:
          - '*'
      - apiGroups:
          - apiextensions.k8s.io
        resources:
          - customresourcedefinitions
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: platform-mesh-kubernetes-graphql-gateway
      namespace: platform-mesh-system
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: platform-mesh-kubernetes-graphql-gateway
    subjects:
      - kind: ServiceAccount
        name: platform-mesh-kubernetes-graphql-gateway
        namespace: platform-mesh-system
